### ISSUE TITLE
TabNavigator slow to change tabs on Android

### ISSUE THREAD

**Issue by sfratini** [react-navigation/react-navigation#4574]
Environment Details:
React Native: 0.48.4
React Navigation: 1.5.11
native-base: 2.4.2

This issue is related to this one, however I believe I have some more info about the problem so I made a new one: https://github.com/react-navigation/react-navigation/issues/739

Behavior:
1) When starting the app, if you click on any footer icon, the first time, the screen loads fast, as expected. 
2) When clicking on the same tab icon again, you get a delay of about 5+ seconds
3) If, on the other hand, you swipe, it also works fast. 

By looking at the other issues, I realized that I was using both Sentry and NativeBase, that people mentioned that was causing problems. I tried removing Sentry but the delay still exists. I also dont have any console logs. 

Debugging:

I made some profiling in React Native and I couldnt find any bottleneck in the JS Thread, just some delay between the execution of some tasks but nothing that was standing out. I also found out that when the actua, TabNavigator code runs in JS it actually runs inmediately. When you swipe, the Tab gets refreshed inmediately and when you click, there is a delay but when the TabNavigator runs, it actually does quickly, so the delay is somewhere else. 

I also found that that in my emulator, I get A LOT of gargabe collector messages right after the first render. 

Then, I switched to some Android native debugging. 

First time clicking on a tab:
![2018-06-25 14_34_42-nebularadio - c__vsts_nebula-showcase_radio-prototipo-app_android - android st](https://user-images.githubusercontent.com/10799414/41872659-edbf3eaa-7898-11e8-9897-0f80f81adbc6.png)

So that is actually two clicks, and CPU stays normal. 

This is the second time I click on that tab (just one click):
![2018-06-25 14_18_18-nebularadio - c__vsts_nebula-showcase_radio-prototipo-app_android - android st](https://user-images.githubusercontent.com/10799414/41872686-0c9f02f6-7899-11e8-9faf-2322cf9c8f7e.png)

![2018-06-25 14_18_57-nebularadio - c__vsts_nebula-showcase_radio-prototipo-app_android - android st](https://user-images.githubusercontent.com/10799414/41872697-159d199c-7899-11e8-90df-2460b2e5bb99.png)

So the second image shows that the CPU and memory runs heavily for several seconds. Since there are a lot of MessageQueue executions, I decided to take a look at the MQ logs. At first, I couldnt run it because Sentry was giving me an OutOfMemory error so I thought I was heavily using the breadcrumb (logs) feature in Sentry but after completely removing the dependency, it does the same. 

One difference I found in the message queue is this:

I/ReactNativeJS: JS->N : UIManager.setChildren([534,[535]])
I/ReactNativeJS: JS->N : UIManager.setChildren([533,[534]])
I/ReactNativeJS: JS->N : UIManager.setChildren([504,[505,513,514,533]])
I/ReactNativeJS: JS->N : UIManager.setChildren([503,[504]])
I/ReactNativeJS: JS->N : UIManager.replaceExistingNonRootView([290,503])

It does the replaceExistingNonRootView like 3 times every time I clic on a single tab, and it does not that the same when I clic the first time. 

I hope this information helps. If there is anything I can test, please let me know, as this bug makes react navigation useless to me. 

Thanks!

**Comment by brentvatne** [comment]
hello! you are using an old version of react-navigation. i'd suggest updating to the latest and trying again. i only handle bug reports against the latest major version of react-navigation so i am going to close this but i want to emphasize that we appreciate detailed reports - the only thing that is missing here is a runnable reproducible example.

**Comment by sfratini** [comment]
@brentvatne Fair enough, I have a quite complex project and creating a MVP is hard enough however I found the issue. It was a mix of native-base footer and my own implementation of a Component which was extending React.Component but also had a lot of extra code. 

Basically I had React.Component -> MyBaseComponent -> FooterTab. 

I replaced my custom tab navigator for a React.PureComponent that only has the code for the Footer and all works well. Just leaving it here in case someone sees this in the future.
